lib/ansible/modules/cloud/amazon/s3_bucket.py

```diff
@@ -244,8 +254,35 @@ def create_or_update_bucket(s3_client, module, location):
             current_tags_dict = tags
             changed = True
 
+    # Encryption
+    try:
+        current_encryption = get_bucket_encryption(s3_client, name)
```

Comment:
You probably want to check if s3_client has the appropriate attribute first, and only call this if it is the case (and only fail if it is not the case and the user has provided the encryption option).
```
hasattr(s3_client, "get_bucket_encryption")
```

Right now if a user has an older version of botocore this will fail with an AttributeError:
```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_s3862i/ansible_module_s3_bucket.py", line 633, in <module>
    main()
  File "/tmp/ansible_s3862i/ansible_module_s3_bucket.py", line 627, in main
    create_or_update_bucket(s3_client, module, location)
  File "/tmp/ansible_s3862i/ansible_module_s3_bucket.py", line 259, in create_or_update_bucket
    current_encryption = get_bucket_encryption(s3_client, name)
  File "/tmp/ansible_s3862i/ansible_modlib.zip/ansible/module_utils/cloud.py", line 153, in retry_func
AttributeError: 'S3' object has no attribute 'get_bucket_encryption'
```